File: Kakao/2018_ n진수 게임.swift
// swift는 진법변환을 직접 구현할 필요가 없음. 

func solution(_ n:Int, _ t:Int, _ m:Int, _ p:Int) -> String {
    var numArray: [Character] = []
    var answer: String = ""
    var num: Int = 0
    while numArray.count < m*t {
        numArray += String(num, radix: n, uppercase: true).map({$0})
        num += 1
    }
    for i in 0..<t {
        answer += String(numArray[m*i+p-1])
    }
    return answer
}

File: Kakao/2018_ n진수 게임.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string encode[6] = {"A","B","C","D","E","F"};

string getNextNumber(int num, int radix)
{
    if(num==0)  return "0"; 
    string number;
    while(num!=0)
    {
        if(num%radix<10)    number = to_string(num%radix) + number;
        else    number = encode[num%radix-10] + number;    
        num /= radix;
    }
    return number;
}

string solution(int n, int t, int m, int p) {
    string answer = "";
    string number;
    int num=0;
    while(number.size()<m*t)
    {
        number += getNextNumber(num++,n);
    }
    for(int i=0;i<t;i++)    answer += number[m*i+(p-1)];
    return answer;
}
